native/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java

```diff
@@ -0,0 +1,621 @@
+     * method to read ballerina files.
+     */
+    public static ArrayList<Entity> readBalFiles(Path filePath, String module) throws IOException {
+        ArrayList<Entity> entityArray = new ArrayList<>();
+        int index = -1;
+        ArrayList<String> keys = new ArrayList<>();
+        String tableName = null;
+        int count;
+        SyntaxTree balSyntaxTree = SyntaxTree.from(TextDocuments.from(Files.readString(filePath)));
+        ModulePartNode rootNote = balSyntaxTree.rootNode();
+        NodeList<ModuleMemberDeclarationNode> nodeList = rootNote.members();
+        for (ModuleMemberDeclarationNode moduleNode : nodeList) {
+            if (moduleNode.kind() != SyntaxKind.TYPE_DEFINITION) {
+                continue;
+            }
+            Collection<ChildNodeEntry> temp = moduleNode.childEntries();
+            for (ChildNodeEntry child : temp) {
+                if (child.name().equals("metadata")) {
+                    if (child.node().isEmpty()) {
+                        continue;
+                    }
+                    MetadataNode metaD = (MetadataNode) child.node().get();
+                    NodeList<AnnotationNode> annotations = metaD.annotations();
+                    for (AnnotationNode annotation : annotations) {
+                        Node annotReference = annotation.annotReference();
+                        if (annotReference.kind() != SyntaxKind.QUALIFIED_NAME_REFERENCE) {
+                            continue;
+                        }
+                        QualifiedNameReferenceNode qualifiedNameRef =
+                                    (QualifiedNameReferenceNode) annotReference;
+                        if (qualifiedNameRef.identifier().text().equals("Entity") && qualifiedNameRef
+                                .modulePrefix().text().equals(BalFileConstants.PERSIST) &&
+                                annotation.annotValue().isPresent()) {
+                            index += 1;
+                            for (MappingFieldNode fieldNode : annotation.annotValue().get().fields()) {
+                                if (fieldNode.kind() != SyntaxKind.SPECIFIC_FIELD) {
+                                    continue;
+                                }
+                                SpecificFieldNode specificField = (SpecificFieldNode) fieldNode;
+                                if (specificField.fieldName().kind() != SyntaxKind.IDENTIFIER_TOKEN ||
+                                        specificField.valueExpr().isEmpty()) {
+                                    continue;
+                                }
+                                ExpressionNode valueNode =
+                                        specificField.valueExpr().get();
+                                if (valueNode instanceof ListConstructorExpressionNode
+                                        && ((IdentifierToken) specificField.fieldName()).text()
+                                        .equals(BalFileConstants.KEY)) {
```

Comment:
```suggestion
                                if (((SpecificFieldNode) fieldNode).fieldName().toString().trim().equals("key")) {
```

Reply:
Updated in [d6a53d3](https://github.com/ballerina-platform/persist-tools/pull/15/commits/d6a53d3f948c8f076acb3cc1f2730258a998cf40)